{doc} Add breaking change rule doc (#26779)
Azure/azure-cli: doc/breaking_change_rules/1005.md

### 1005 - Cmd Property Updated

**Description**: Checks whether a cmd's property value is updated from the previous cli version. 

> **Analysis**: Not all configured properties updated for a cmd is considered as breaking change. The properties listed in CMD_PROPERTY_UPDATE_BREAK_LIST below are thought to influence or interrupt current workflow and be breaking changes.

Reviewer: ```suggestion
**Analysis**: Not all configured properties updated for a cmd are considered breaking change. The properties listed in CMD_PROPERTY_UPDATE_BREAK_LIST below are thought to influence or interrupt current workflow and be breaking changes.
```

> For now, updated properties in cmd level do not cause breaking workflows.

Reviewer: ```suggestion
For now, updated properties in cmd level does not break workflow.
```
